sdk/python/feast/data_source.py

```diff
@@ -297,6 +297,7 @@ def from_proto(data_source: DataSourceProto) -> Any:
             raise ValueError("Could not identify the source type being added.")
 
         if data_source_type == DataSourceProto.SourceType.CUSTOM_SOURCE:
+            data_source.data_source_class_type = "feast.infra.offline_stores.contrib.mssql_offline_store.mssqlserver_source.MsSqlServerSource"
             cls = get_data_source_class_from_type(data_source.data_source_class_type)
```

Comment:
Instead of hard coding the class type here, I believe it should be declared in the to_proto() function of the MsSqlServerSource class. This is done in case of the other offline contrib stores and also ensures conflicts with any other future custom sources.

Reply:
fixed